File: csharp/src/Models/Earring.cs
using NecklaceRefactoringKata.Enums;

namespace NecklaceRefactoringKata.Models
{
    public record Earring(EarringType Type, Jewel Stone) : Jewellery(Stone, IsSmall: Type == EarringType.Stud, IsEarring: true);
}

File: php/src/Earring.php
<?php

namespace NecklaceRefactoringKata;

enum EarringType {
    case Stud;
    case Hoop;
    case Drop;
}

class Earring extends Jewellery
{
    public function __construct(public EarringType $type)
    {
    }

    public function isSmall(): bool
    {
        return $this->type === EarringType::Stud;
    }

    public function isEarring(): bool
    {
        return true;
    }
}

